plugin/storage/cassandra/spanstore/query_builder_test.go

```diff
@@ -64,101 +65,101 @@ func TestGoodQueryScenarios(t *testing.T) {
 		{
 			&spanstore.TraceQueryParameters{NumTraces: 25},
 			expectedQueries{
-				mainQuery:       "SELECT trace_id, span_id FROM traces ALLOW FILTERING",
-				mainQueryParams: nil,
+				mainQuery:       "SELECT trace_id, span_id FROM traces LIMIT ? ALLOW FILTERING",
```

Comment:
Two reasons:
1. When doing a client-side join with tags, adding a limit does not respect the user-defined limit value
2. We are returning unique spans, but limit is defined as number of traces.

I still think we should put limit in here, it's a big performance improvemnt.